### ISSUE TITLE
Rigaku/Bluesky Does Not Finish 10,000 Measurement

### ISSUE THREAD

**Issue by qzhang234** [aps-8id-dys/ipython-8idiuser#223]
Bluesky jammed at 8209/10000. Attached are the last lines of the logger file, located at:
`/home/beams/8IDIUSER/bluesky_data/2020/2020-3/.logs/`.
Looks like there's something wrong with the timing of kicking off DM workflow.

<details>
<summary> last lines of Bluesky logger </summary>

```
|2020-09-12 12:11:05.634|INFO|7639|bluesky-session|xpcs_acquire|281|bluesky-run-engine| - scaler should be autocounting now
|2020-09-12 12:11:05.862|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo FILE:F:A166_Test_att00_08203 | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=32, #connections=11, #context switches=2]
|2020-09-12 12:11:06.510|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=33, #connections=11, #context switches=2]
|2020-09-12 12:11:10.155|INFO|7639|bluesky-session|xpcs_acquire|307|Thread-755555| - DM workflow kickoff starting: analysis:True  file:/home/8-id-i/2020-3/test202008/A166_Test_att00_08203/A166_Test_att00_08203_0001-100000.hdf
|2020-09-12 12:11:10.391|INFO|7639|bluesky-session|bp_Test_Bluesky|59|bluesky-run-engine| - A166_Test_att00_08204
|2020-09-12 12:11:10.391|INFO|7639|bluesky-session|xpcs_acquire|48|bluesky-run-engine| - AD_Acquire starting
|2020-09-12 12:11:10.391|INFO|7639|bluesky-session|xpcs_acquire|59|bluesky-run-engine| - file_path = /home/8-id-i/2020-3/test202008/A166_Test_att00_08204/
|2020-09-12 12:11:10.442|INFO|7639|bluesky-session|xpcs_acquire|317|bluesky-run-engine| - calling full_acquire_procedure()
|2020-09-12 12:11:10.451|INFO|7639|bluesky-session|xpcs_acquire|312|Thread-755555| - DM workflow kickoff done
|2020-09-12 12:11:10.451|INFO|7639|bluesky-session|xpcs_acquire|313|Thread-755555| - b'id=1297cb8e-593c-4166-8eca-5d5217f90973 owner=8idiuser status=pending startTime=1599930670.43 startTimestamp=2020/09/12 12:11:10 CDT \n'
|2020-09-12 12:11:10.453|INFO|7639|bluesky-session|xpcs_acquire|146|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-09-12 12:11:10.489|INFO|7639|bluesky-session|xpcs_acquire|281|bluesky-run-engine| - scaler should be autocounting now
|2020-09-12 12:11:10.756|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo FILE:F:A166_Test_att00_08204 | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=32, #connections=11, #context switches=2]
|2020-09-12 12:11:11.305|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=33, #connections=11, #context switches=2]
|2020-09-12 12:11:15.169|INFO|7639|bluesky-session|xpcs_acquire|307|Thread-755647| - DM workflow kickoff starting: analysis:True  file:/home/8-id-i/2020-3/test202008/A166_Test_att00_08204/A166_Test_att00_08204_0001-100000.hdf
|2020-09-12 12:11:15.443|INFO|7639|bluesky-session|bp_Test_Bluesky|59|bluesky-run-engine| - A166_Test_att00_08205
|2020-09-12 12:11:15.443|INFO|7639|bluesky-session|xpcs_acquire|48|bluesky-run-engine| - AD_Acquire starting
|2020-09-12 12:11:15.444|INFO|7639|bluesky-session|xpcs_acquire|59|bluesky-run-engine| - file_path = /home/8-id-i/2020-3/test202008/A166_Test_att00_08205/
|2020-09-12 12:11:15.474|INFO|7639|bluesky-session|xpcs_acquire|312|Thread-755647| - DM workflow kickoff done
|2020-09-12 12:11:15.474|INFO|7639|bluesky-session|xpcs_acquire|313|Thread-755647| - b'id=78117c74-9950-4f14-911e-50d63171c96a owner=8idiuser status=pending startTime=1599930675.46 startTimestamp=2020/09/12 12:11:15 CDT \n'
|2020-09-12 12:11:15.501|INFO|7639|bluesky-session|xpcs_acquire|317|bluesky-run-engine| - calling full_acquire_procedure()
|2020-09-12 12:11:15.546|INFO|7639|bluesky-session|xpcs_acquire|146|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-09-12 12:11:15.588|INFO|7639|bluesky-session|xpcs_acquire|281|bluesky-run-engine| - scaler should be autocounting now
|2020-09-12 12:11:15.815|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo FILE:F:A166_Test_att00_08205 | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=32, #connections=11, #context switches=2]
|2020-09-12 12:11:16.369|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=33, #connections=11, #context switches=2]
|2020-09-12 12:11:20.098|INFO|7639|bluesky-session|xpcs_acquire|307|Thread-755739| - DM workflow kickoff starting: analysis:True  file:/home/8-id-i/2020-3/test202008/A166_Test_att00_08205/A166_Test_att00_08205_0001-100000.hdf
|2020-09-12 12:11:20.341|INFO|7639|bluesky-session|bp_Test_Bluesky|59|bluesky-run-engine| - A166_Test_att00_08206
|2020-09-12 12:11:20.342|INFO|7639|bluesky-session|xpcs_acquire|48|bluesky-run-engine| - AD_Acquire starting
|2020-09-12 12:11:20.342|INFO|7639|bluesky-session|xpcs_acquire|59|bluesky-run-engine| - file_path = /home/8-id-i/2020-3/test202008/A166_Test_att00_08206/
|2020-09-12 12:11:20.363|INFO|7639|bluesky-session|xpcs_acquire|317|bluesky-run-engine| - calling full_acquire_procedure()
|2020-09-12 12:11:20.373|INFO|7639|bluesky-session|xpcs_acquire|146|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-09-12 12:11:20.384|INFO|7639|bluesky-session|xpcs_acquire|312|Thread-755739| - DM workflow kickoff done
|2020-09-12 12:11:20.384|INFO|7639|bluesky-session|xpcs_acquire|313|Thread-755739| - b'id=d59fe7df-2043-45c4-bc8f-a76ed2b5c8f4 owner=8idiuser status=pending startTime=1599930680.37 startTimestamp=2020/09/12 12:11:20 CDT \n'
|2020-09-12 12:11:20.439|INFO|7639|bluesky-session|xpcs_acquire|281|bluesky-run-engine| - scaler should be autocounting now
|2020-09-12 12:11:20.765|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo FILE:F:A166_Test_att00_08206 | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=32, #connections=11, #context switches=2]
|2020-09-12 12:11:21.333|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=33, #connections=11, #context switches=2]
|2020-09-12 12:11:25.202|INFO|7639|bluesky-session|xpcs_acquire|307|Thread-755831| - DM workflow kickoff starting: analysis:True  file:/home/8-id-i/2020-3/test202008/A166_Test_att00_08206/A166_Test_att00_08206_0001-100000.hdf
|2020-09-12 12:11:25.441|INFO|7639|bluesky-session|bp_Test_Bluesky|59|bluesky-run-engine| - A166_Test_att00_08207
|2020-09-12 12:11:25.441|INFO|7639|bluesky-session|xpcs_acquire|48|bluesky-run-engine| - AD_Acquire starting
|2020-09-12 12:11:25.442|INFO|7639|bluesky-session|xpcs_acquire|59|bluesky-run-engine| - file_path = /home/8-id-i/2020-3/test202008/A166_Test_att00_08207/
|2020-09-12 12:11:25.461|INFO|7639|bluesky-session|xpcs_acquire|317|bluesky-run-engine| - calling full_acquire_procedure()
|2020-09-12 12:11:25.471|INFO|7639|bluesky-session|xpcs_acquire|146|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-09-12 12:11:25.505|INFO|7639|bluesky-session|xpcs_acquire|312|Thread-755831| - DM workflow kickoff done
|2020-09-12 12:11:25.505|INFO|7639|bluesky-session|xpcs_acquire|313|Thread-755831| - b'id=774c85d9-0144-4b7a-b171-8229c300cdc4 owner=8idiuser status=pending startTime=1599930685.49 startTimestamp=2020/09/12 12:11:25 CDT \n'
|2020-09-12 12:11:25.532|INFO|7639|bluesky-session|xpcs_acquire|281|bluesky-run-engine| - scaler should be autocounting now
|2020-09-12 12:11:25.820|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo FILE:F:A166_Test_att00_08207 | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=32, #connections=11, #context switches=2]
|2020-09-12 12:11:26.356|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=33, #connections=11, #context switches=2]
|2020-09-12 12:11:30.081|INFO|7639|bluesky-session|xpcs_acquire|307|Thread-755923| - DM workflow kickoff starting: analysis:True  file:/home/8-id-i/2020-3/test202008/A166_Test_att00_08207/A166_Test_att00_08207_0001-100000.hdf
|2020-09-12 12:11:30.340|INFO|7639|bluesky-session|bp_Test_Bluesky|59|bluesky-run-engine| - A166_Test_att00_08208
|2020-09-12 12:11:30.340|INFO|7639|bluesky-session|xpcs_acquire|48|bluesky-run-engine| - AD_Acquire starting
|2020-09-12 12:11:30.340|INFO|7639|bluesky-session|xpcs_acquire|59|bluesky-run-engine| - file_path = /home/8-id-i/2020-3/test202008/A166_Test_att00_08208/
|2020-09-12 12:11:30.362|INFO|7639|bluesky-session|xpcs_acquire|317|bluesky-run-engine| - calling full_acquire_procedure()
|2020-09-12 12:11:30.368|INFO|7639|bluesky-session|xpcs_acquire|312|Thread-755923| - DM workflow kickoff done
|2020-09-12 12:11:30.368|INFO|7639|bluesky-session|xpcs_acquire|313|Thread-755923| - b'id=e7f801e8-c68a-4375-a885-f3ec27fbd38b owner=8idiuser status=pending startTime=1599930690.35 startTimestamp=2020/09/12 12:11:30 CDT \n'
|2020-09-12 12:11:30.372|INFO|7639|bluesky-session|xpcs_acquire|146|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-09-12 12:11:30.441|INFO|7639|bluesky-session|xpcs_acquire|281|bluesky-run-engine| - scaler should be autocounting now
|2020-09-12 12:11:30.726|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo FILE:F:A166_Test_att00_08208 | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=32, #connections=11, #context switches=2]
|2020-09-12 12:11:31.428|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=33, #connections=11, #context switches=2]
|2020-09-12 12:11:35.098|INFO|7639|bluesky-session|xpcs_acquire|307|Thread-756015| - DM workflow kickoff starting: analysis:True  file:/home/8-id-i/2020-3/test202008/A166_Test_att00_08208/A166_Test_att00_08208_0001-100000.hdf
|2020-09-12 12:11:35.345|INFO|7639|bluesky-session|bp_Test_Bluesky|59|bluesky-run-engine| - A166_Test_att00_08209
|2020-09-12 12:11:35.345|INFO|7639|bluesky-session|xpcs_acquire|48|bluesky-run-engine| - AD_Acquire starting
|2020-09-12 12:11:35.346|INFO|7639|bluesky-session|xpcs_acquire|59|bluesky-run-engine| - file_path = /home/8-id-i/2020-3/test202008/A166_Test_att00_08209/
|2020-09-12 12:11:35.367|INFO|7639|bluesky-session|xpcs_acquire|317|bluesky-run-engine| - calling full_acquire_procedure()
|2020-09-12 12:11:35.376|INFO|7639|bluesky-session|xpcs_acquire|146|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-09-12 12:11:35.396|INFO|7639|bluesky-session|xpcs_acquire|312|Thread-756015| - DM workflow kickoff done
|2020-09-12 12:11:35.397|INFO|7639|bluesky-session|xpcs_acquire|313|Thread-756015| - b'id=9ea3b756-f537-4617-b159-b317ba042743 owner=8idiuser status=pending startTime=1599930695.38 startTimestamp=2020/09/12 12:11:35 CDT \n'
|2020-09-12 12:11:35.443|INFO|7639|bluesky-session|xpcs_acquire|281|bluesky-run-engine| - scaler should be autocounting now
|2020-09-12 12:11:35.721|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo FILE:F:A166_Test_att00_08209 | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=32, #connections=11, #context switches=2]
|2020-09-12 12:11:36.268|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=33, #connections=11, #context switches=2]
|2020-09-13 00:09:14.534|INFO|7639|bluesky-session|bp_Test_Bluesky|59|bluesky-run-engine| - A167_Test_att00_00001
|2020-09-13 00:09:14.535|INFO|7639|bluesky-session|xpcs_acquire|48|bluesky-run-engine| - AD_Acquire starting
|2020-09-13 00:09:14.535|INFO|7639|bluesky-session|xpcs_acquire|59|bluesky-run-engine| - file_path = /home/8-id-i/2020-3/test202008/A167_Test_att00_00001/
|2020-09-13 00:09:14.563|INFO|7639|bluesky-session|xpcs_acquire|317|bluesky-run-engine| - calling full_acquire_procedure()
|2020-09-13 00:09:14.574|INFO|7639|bluesky-session|xpcs_acquire|146|bluesky-run-engine| - detNum=46, det_pars={'ccdHardwareColSize': 512, 'ccdHardwareRowSize': 1024, 'ccdxsense': 1, 'ccdzsense': 1, 'harmonic': 1, 'dpix': 0.076, 'saturation': 3, 'adupphot': 1.0, 'efficiency': 1.0, 'gain': 1, 'blemish': 1, 'flatfield': 0, 'distortion': 0, 'parasitic': 0, 'lld': 0, 'manufacturer': 'RIGAKU500K_NoGap'}
|2020-09-13 00:09:14.660|INFO|7639|bluesky-session|xpcs_acquire|281|bluesky-run-engine| - scaler should be autocounting now
|2020-09-13 00:09:15.036|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo FILE:F:A167_Test_att00_00001 | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=33, #connections=11, #context switches=2]
|2020-09-13 00:09:15.467|INFO|7639|bluesky-session|rigaku_ufxc|63|bluesky-run-engine| - echo EXPOSURE | nc rigaku1.xray.aps.anl.gov 10000 [#open files=4, #open fd=41, #threads=34, #connections=11, #context switches=2]
|2020-09-13 00:09:19.284|INFO|7639|bluesky-session|xpcs_acquire|307|Thread-756231| - DM workflow kickoff starting: analysis:True  file:/home/8-id-i/2020-3/test202008/A167_Test_att00_00001/A167_Test_att00_00001_0001-100000.hdf

```

</details>

The logger file is attached:
[ipython_console.log](https://github.com/aps-8id-dys/ipython-8idiuser/files/5213659/ipython_console.log)

**Comment by qzhang234** [comment]
Run status:

<img width="1259" alt="Screen Shot 2020-09-12 at 4 15 15 PM" src="https://user-images.githubusercontent.com/48140482/93010732-32b57700-f555-11ea-87e6-e025eb31df95.png">
<img width="1251" alt="Screen Shot 2020-09-12 at 4 15 24 PM" src="https://user-images.githubusercontent.com/48140482/93010736-36e19480-f555-11ea-9dfc-3d08fc750bbf.png">
<img width="885" alt="Screen Shot 2020-09-12 at 4 17 09 PM" src="https://user-images.githubusercontent.com/48140482/93010737-377a2b00-f555-11ea-9549-cbfda8a1e99b.png">

**Comment by qzhang234** [comment]
Hard disk status. `/home/8-id-i/` is almost full but still has 400 GB or so space:

<img width="709" alt="Screen Shot 2020-09-12 at 4 16 23 PM" src="https://user-images.githubusercontent.com/48140482/93010741-495bce00-f555-11ea-9d0b-d5664f3dfe36.png">

**Comment by qzhang234** [comment]
Upon restarting the run, the scan continued, which indicate the previous jam was not a hard disk volume issue.

<details>
<summary> screen output, plan restarted </summary>

```

560a0117-2760-4162-b856-ea28de5ba7c3',
 '4245e7c4-ebd0-48a8-bbbe-4f9cb3679a26',
 '455f62f2-8dc5-4726-a4ac-87d150a7e673',
 '567f817a-c390-4dd6-a583-aba9f0023315',
 '78ffdfcd-c158-49f1-9a65-1554dde37cd7',
 '5e6450d1-c7e5-4916-94c3-e843b3118996',
 '3c3f69f2-6d14-4e7d-9ef4-67bd01956750',
 '5577a0f5-cce6-4142-82b6-0225b7eed79a',
 '832bef5b-3a2d-4518-b35c-2581aea99155',
 'f16ed0cb-ef55-4f11-b928-3909d5be3243',
 'bd2cc992-106e-4396-9f6b-24a3fc222eef',
 ...)

In [12]: 

In [12]: 

In [12]: 

In [12]: 

In [12]: RE(multi_capil())
A167_Test_att00_00001                                                                                                                         
HDF5 workflow file name: /home/8-id-i/2020-3/test202008/A167_Test_att00_00001/A167_Test_att00_00001_0001-100000.hdf                           
A167_Test_att00_00002
HDF5 workflow file name: /home/8-id-i/2020-3/test202008/A167_Test_att00_00002/A167_Test_att00_00002_0001-100000.hdf                           
A167_Test_att00_00003
HDF5 workflow file name: /home/8-id-i/2020-3/test202008/A167_Test_att00_00003/A167_Test_att00_00003_0001-100000.hdf                           
A167_Test_att00_00004
^CA 'deferred pause' has been requested. The RunEngine will pause at the next checkpoint. To pause immediately, hit Ctrl+C again in the next 10 seconds.
Deferred pause acknowledged. Continuing to checkpoint.
ritrying a second timeo progress bar available.]                                                                                              
Pausing...
---------------------------------------------------------------------------                                                                   
RunEngineInterrupted                      Traceback (most recent call last)
<ipython-input-12-3d05a9955984> in <module>
----> 1 RE(multi_capil())

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in __call__(self, *args, **metadata_kw)
    808 
    809         if self._interrupted:
--> 810             raise RunEngineInterrupted(self.pause_msg) from None
    811 
    812         return tuple(self._run_start_uids)

RunEngineInterrupted: 
Your RunEngine is entering a paused state. These are your options for changing
the state of the RunEngine:

RE.resume()    Resume the plan.
RE.abort()     Perform cleanup, then kill plan. Mark exit_stats='aborted'.
RE.stop()      Perform cleanup, then kill plan. Mark exit_status='success'.
RE.halt()      Emergency Stop: Do not perform cleanup --- just stop.


In [13]: RE.abort()
Aborting: running cleanup and marking exit_status as 'abort'...
Out[13]: 
('b2c68be4-0c47-4b21-a636-166bbe654820',
 '1b09b066-ab3a-4386-9f58-ba3270534822',
 'a872cf5c-c0ce-4cf1-9aa3-589d099c0c9b',
 'd5da87ef-e157-40bd-88b9-94047775d31b')

In [14]: 

```

</details>

<img width="1072" alt="Screen Shot 2020-09-13 at 12 11 26 AM" src="https://user-images.githubusercontent.com/48140482/93010788-b1aaaf80-f555-11ea-8369-ff2e3de747b6.png">

**Comment by qzhang234** [comment]
Attached is the log on Rigaku Labview panel. The regions highlighted by the red rectangle is different at the scan number where the scan crashed.

<img width="471" alt="Screen Shot 2020-09-13 at 12 17 40 AM" src="https://user-images.githubusercontent.com/48140482/93010892-bb80e280-f556-11ea-8a19-387c7ac4c0f9.png">

**Comment by qzhang234** [comment]
The second test finished 10000/10000 measurement! 

I'll submit another 10000 to test again.

<img width="1382" alt="Screen Shot 2020-09-13 at 3 30 54 PM" src="https://user-images.githubusercontent.com/48140482/93028182-b617ac80-f5d7-11ea-8b86-651f52f9e60c.png">

**Comment by prjemian** [comment]
This is a lot to digest on a beautiful Sunday afternoon.  Is it possible that a new DM workflow was started before the old one had either finished or even started?  What is the problem to be solved?

**Comment by prjemian** [comment]
I'm encouraged that two consecutive sequences of 10k runs each were successful.  How about workflow execution time.  Is that still increasing?  Does it reset with a new sequence of 10k runs?

**Comment by qzhang234** [comment]
The workflow execution time is not increasing. 

I guess the problem to be solved is to figure out why the first 10000 measurement (A166) crashed at 8209/10000. I can see from the Bluesky log and the Rigaku Labview panel that A166_8209 is different from the rest, but I do not know what the messages mean. 

I'm waiting for the Rigaku people to wake up and explain to me what the log indicates, so in the meantime if you could look at the Bluesky log then we could try to come up with a solution when we meet on Tuesday

**Comment by qzhang234** [comment]
I started the third test, X003. Let's see if we could reproduce the jam in A166.

<img width="1288" alt="Screen Shot 2020-09-13 at 5 11 02 PM" src="https://user-images.githubusercontent.com/48140482/93029873-29272000-f5e4-11ea-9429-d55c2d0f2ee9.png">

**Comment by prjemian** [comment]
Looking at the various log reports above

* it is obvious that the problem happened during scan A166_8209 as you noted.  
* The `ipython_console.log` provides nothing useful here and can be ignored.  
* The _last lines of Bluesky logger_ show that:
  - run 8209 started
  - the DM workflow for 8208 completed
  - the Rigaku detector was sent the EXPOSURE command
  - the DM workflow for 9209 was never started
* The console window snapshot shows a last line of `rigaku [In progress. No progress bar available]`.  
  - This means the rigaku detector has been triggered and Bluesky can't draw a progress bar.
  - This is normal for bluesky since it has no idea of when the trigger will be complete nor does it have any interim updates from the detector.  
  -  It is not normal for this text to be on display for more than a few seconds.  Once the trigger is complete, this line will be cleared.
  - Previous logs show that a little more than 4 seconds is typical between starting the Rigaku detector and starting a DM workflow.
* We do not log (now) the sequence of events which trigger the Rigaku and when the trigger is done.

The Rigaku detector support has these attributes:

```
    acquire_start = Component(EpicsSignal, "8idi:Unidig2Bo7.VAL")
    acquire_complete = Component(EpicsSignalRO, "8idi:Unidig2Bi2.VAL")
```

This is the trigger code: https://github.com/aps-8id-dys/ipython-8idiuser/blob/1d91a4e731f5977c149a6513ff916c35be23b02e/profile_bluesky/startup/instrument/devices/rigaku_ufxc.py#L196-L229

Line 215 shows the EXPOSURE command sent to the detector (this line shows up in the logs at 2020-09-12 12:11:36.268) and is the last piece of diagnostic information we have from Bluesky.

I don't understand the marked regions of the _Rigaku Labview panel_ and why they might be unusual.

Everything I've noted points to some problem with (or just after) triggering the Rigaku detector during 9209.  Bluesky is waiting for `acquire_complete` (`8idi:Unidig2Bi2.VAL`) to go from 0 to 1.

**Comment by qzhang234** [comment]
@prjemian Combining what you said (Bluesky is waiting for `acquire_complete` (`8idi:Unidig2Bi2.VAL`) to go from 0 to 1), and Nakaye's past explanation about the meaning of the s0\s1 (Forward slash is the Yen symbol￥ by the way and I absolutely hate it), I think you are right: 

At 17:11:03:10 (probably Japanese time), Rigaku sent out 'shutter close and ready' signal (s0\s1). However for some reason that signal was flipped to `low` after only 10 ms. This is different from all the other successful measurements. As a result Bluesky kept waiting for the ready signal from Rigaku, which seems like it never came.

I'll inquire the Rigaku team further about what the cause of this problem could be

<img width="634" alt="Screen Shot 2020-09-13 at 9 56 11 PM" src="https://user-images.githubusercontent.com/48140482/93038969-f7c24a80-f60b-11ea-88b0-3603a44cb465.png">

**Comment by qzhang234** [comment]
@prjemian The third test submitted on Sunday afternoon crashed at 3894/10000. Any suggestions what the error message means?

<img width="1243" alt="Screen Shot 2020-09-14 at 10 57 03 AM" src="https://user-images.githubusercontent.com/48140482/93108969-1579c900-f679-11ea-9006-83ea133055cf.png">

<details>
<summary> Error message on Bluesky terminal </summary>

```

X003_Test_att00_03893
HDF5 workflow file name: /home/8-id-i/2020-3/test202008/X003_Test_att00_03893/X003_Test_att00_03893_0001-100000.hdf                           
X003_Test_att00_03894
HDF5 workflow file name: /home/8-id-i/2020-3/test202008/X003_Test_att00_03894/X003_Test_att00_03894_0001-100000.hdf                           
X003_Test_att00_03895
---------------------------------------------------------------------------                                                                   
DmException                               Traceback (most recent call last)
<ipython-input-18-e00c4c348150> in <module>
----> 1 _ = RE(multi_capil())

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in __call__(self, *args, **metadata_kw)
    805             self._task_fut.add_done_callback(set_blocking_event)
    806 
--> 807         self._resume_task(init_func=_build_task)
    808 
    809         if self._interrupted:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _resume_task(self, init_func)
    929                     if (exc is not None
    930                             and not isinstance(exc, _RunEnginePanic)):
--> 931                         raise exc
    932 
    933     def install_suspender(self, suspender):

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _run(self)
   1498             exit_reason = str(err)
   1499             self.log.exception("Run aborted")
-> 1500             raise err
   1501         finally:
   1502             if not exit_reason:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _run(self)
   1363                     else:
   1364                         try:
-> 1365                             msg = self._plan_stack[-1].send(resp)
   1366                         # We have exhausted the top generator
   1367                         except StopIteration:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in __call__(self, plan)
   1305         plan = monitor_during_wrapper(plan, self.monitors)
   1306         plan = baseline_wrapper(plan, self.baseline)
-> 1307         return (yield from plan)
   1308 
   1309 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in baseline_wrapper(plan, devices, name)
   1160         return (yield from plan)
   1161     else:
-> 1162         return (yield from plan_mutator(plan, insert_baseline))
   1163 
   1164 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in monitor_during_wrapper(plan, signals)
    801     plan1 = plan_mutator(plan, insert_after_open)
    802     plan2 = plan_mutator(plan1, insert_before_close)
--> 803     return (yield from plan2)
    804 
    805 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in fly_during_wrapper(plan, flyers)
    859     plan1 = plan_mutator(plan, insert_after_open)
    860     plan2 = plan_mutator(plan1, insert_before_close)
--> 861     return (yield from plan2)
    862 
    863 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    168                     continue
    169                 else:
--> 170                     raise ex
    171         # if inserting / mutating, put new generator on the stack
    172         # and replace the current msg with the first element from the

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    121             ret = result_stack.pop()
    122             try:
--> 123                 msg = plan_stack[-1].send(ret)
    124             except StopIteration as e:
    125                 # discard the exhausted generator

/home/beams10/8IDIUSER/bluesky_data/2020/2020-3/bp_Test_Bluesky.py in multi_capil()
     91     z0 = 0
     92 
---> 93     yield from repeat_acq(sample_name='Test',att_value=0,acq_rep=10000,samx_cen=4.94,samz_cen=z0)
     94 
     95     # yield from repeat_acq(sample_name='D100_NoFly',att_value=2,acq_rep=5,samx_cen=9.4,samz_cen=z0)

/home/beams10/8IDIUSER/bluesky_data/2020/2020-3/bp_Test_Bluesky.py in repeat_acq(areadet, file_header, acq_rep, sample_name, Temp_list, att_value, FLY_YES_NO, samx_cen, samz_cen)
     65         print(acq_name)
     66 
---> 67         yield from AD_Acquire(
     68             rigaku,
     69             acquire_time=19.00e-6,

~/.ipython-bluesky/profile_bluesky/startup/instrument/plans/xpcs_acquire.py in AD_Acquire(areadet, file_name, acquire_time, acquire_period, num_images, path, submit_xpcs_job, atten, md)
    316 
    317     logger.info("calling full_acquire_procedure()")
--> 318     return (yield from full_acquire_procedure(md=md))

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/utils.py in dec_inner(*inner_args, **inner_kwargs)
   1112                 plan = gen_func(*inner_args, **inner_kwargs)
   1113                 plan = wrapper(plan, *args, **kwargs)
-> 1114                 return (yield from plan)
   1115             return dec_inner
   1116         return dec

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in stage_wrapper(plan, devices)
    950         return (yield from plan)
    951 
--> 952     return (yield from finalize_wrapper(inner(), unstage_devices()))
    953 
    954 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in finalize_wrapper(plan, final_plan, pause_for_debug)
    507     cleanup = True
    508     try:
--> 509         ret = yield from plan
    510     except GeneratorExit:
    511         cleanup = False

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in inner()
    948     def inner():
    949         yield from stage_devices()
--> 950         return (yield from plan)
    951 
    952     return (yield from finalize_wrapper(inner(), unstage_devices()))

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/utils.py in dec_inner(*inner_args, **inner_kwargs)
   1112                 plan = gen_func(*inner_args, **inner_kwargs)
   1113                 plan = wrapper(plan, *args, **kwargs)
-> 1114                 return (yield from plan)
   1115             return dec_inner
   1116         return dec

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in monitor_during_wrapper(plan, signals)
    801     plan1 = plan_mutator(plan, insert_after_open)
    802     plan2 = plan_mutator(plan1, insert_before_close)
--> 803     return (yield from plan2)
    804 
    805 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.ipython-bluesky/profile_bluesky/startup/instrument/plans/xpcs_acquire.py in full_acquire_procedure(md)
    284         logger.debug("before count()")
    285         # yield from bp.count([areadet], md=md)
--> 286         yield from inner_count([areadet], md=_md)
    287         logger.debug("after count()")
    288 

~/.ipython-bluesky/profile_bluesky/startup/instrument/plans/xpcs_acquire.py in inner_count(devices, md)
    258         for obj in devices:
    259             yield from bps.unstage(obj)
--> 260         yield from bps.close_run()
    261         # return ret
    262 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/plan_stubs.py in close_run(exit_status, reason)
    841     :func:`bluesky.plans_stubs.open_run`
    842     """
--> 843     return (yield Msg('close_run', exit_status=exit_status, reason=reason))
    844 
    845 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in new_gen()
    793             def new_gen():
    794                 yield from ensure_generator(unmonitor_msgs)
--> 795                 yield msg
    796             return new_gen(), None
    797         else:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in new_gen()
    851                 yield from ensure_generator(complete_msgs)
    852                 yield from ensure_generator(collect_msgs)
--> 853                 yield msg
    854             return new_gen(), None
    855         else:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in new_gen()
    793             def new_gen():
    794                 yield from ensure_generator(unmonitor_msgs)
--> 795                 yield msg
    796             return new_gen(), None
    797         else:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
     76             # if we have a stashed exception, pass it along
     77             try:
---> 78                 msg = plan_stack[-1].throw(exception)
     79             except StopIteration as e:
     80                 # discard the exhausted generator

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in post_baseline()
   1149         elif msg.command == 'close_run':
   1150             def post_baseline():
-> 1151                 yield from trigger_and_read(devices, name=name)
   1152                 return (yield msg)
   1153 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/plan_stubs.py in trigger_and_read(devices, name)
    909         return ret
    910     from .preprocessors import rewindable_wrapper
--> 911     return (yield from rewindable_wrapper(inner_trigger_and_read(),
    912                                           rewindable))
    913 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in rewindable_wrapper(plan, rewindable)
    691                                             restore_rewindable()))
    692     else:
--> 693         return (yield from plan)
    694 
    695 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/plan_stubs.py in inner_trigger_and_read()
    903         ret = {}  # collect and return readings to give plan access to them
    904         for obj in devices:
--> 905             reading = (yield from read(obj))
    906             if reading is not None:
    907                 ret.update(reading)

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/plan_stubs.py in read(obj)
     93         Msg('read', obj)
     94     """
---> 95     return (yield Msg('read', obj))
     96 
     97 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    193         try:
    194             # yield out the 'current message' and collect the return
--> 195             inner_ret = yield msg
    196         except GeneratorExit:
    197             # special case GeneratorExit.  We must clean up all of our plans

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _run(self)
   1423                         # exceptions (coming in via throw) can be
   1424                         # raised
-> 1425                         new_response = await coro(msg)
   1426 
   1427                     # special case `CancelledError` and let the outer

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/bluesky/run_engine.py in _read(self, msg)
   1668         obj = msg.obj
   1669         # actually _read_ the object
-> 1670         ret = obj.read(*msg.args, **msg.kwargs)
   1671 
   1672         if ret is None:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/ophyd/device.py in read(self)
   1235 
   1236         for _, component in self._get_components_of_kind(Kind.normal):
-> 1237             res.update(component.read())
   1238         return res
   1239 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/ophyd/utils/epics_pvs.py in wrapper(self, *args, **kwargs)
    196     def wrapper(self, *args, **kwargs):
    197         if self.connected:
--> 198             return fcn(self, *args, **kwargs)
    199         else:
    200             raise DisconnectedError('{} is not connected'.format(self.name))

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/ophyd/signal.py in read(self)
    364             dict
    365         '''
--> 366         value = self.get()
    367         return {self.name: {'value': value,
    368                             'timestamp': self.timestamp}}

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/apstools/devices.py in get(self)
    193         if time.time() > self._time_last_updated + self._update_period_s:
    194             # only update from data management once per day
--> 195             self._cycle_name = getCurrentCycle()
    196             self._time_updated = time.time()
    197             # TODO: could get the end time for current cycle and use that instead

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/apstools/beamtime/apsbss.py in getCurrentCycle()
    279 def getCurrentCycle():
    280     """Return the name of the current APS run cycle."""
--> 281     return api_bss.getCurrentRun()["name"]
    282 
    283 

<decorator-gen-950> in getCurrentRun(self)

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/dm/common/api/dmApi.py in wrappedCall(func, *args, **kwargs)
     45             def wrappedCall(func, *args, **kwargs):
     46                 try:
---> 47                     response = func(*args, **kwargs)
     48                     return response
     49                 except DmException as ex:

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/dm/aps_db_web_service/api/bssApsDbApi.py in getCurrentRun(self)
    113         '''
    114         url = '%s/runs/current' % (self.getContextRoot())
--> 115         responseDict = self.sendRequest(url=url, method='GET')
    116         return RunInfo(responseDict)
    117 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/dm/common/api/dmRestApi.py in sendRequest(self, url, method, contentType, data)
    143         # setHost()
    144         sm.setHost(self.__getWebServiceUrl(url))
--> 145         (response, responseData) = self.getSessionManager().sendRequest(url, method, contentType, data)
    146         return json.loads(responseData)
    147 

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/dm/common/client/sessionManager.py in sendRequest(self, url, method, contentType, data)
    215 
    216         # Check headers for errors and update session cookie
--> 217         sessionCookie = self.checkResponseHeadersForErrorsAndSaveSession(response.headers)
    218         if sessionCookie != None:
    219             self.sessionCookie = sessionCookie

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/dm/common/client/sessionManager.py in checkResponseHeadersForErrorsAndSaveSession(self, responseHeaders)
    234     def checkResponseHeadersForErrorsAndSaveSession(self, responseHeaders):
    235         try:
--> 236             DmExceptionMapper.checkStatus(responseHeaders)
    237             sessionCookie = responseHeaders.get('Set-Cookie')
    238             self.saveSession(sessionCookie)

~/.conda/envs/bluesky_2020_9/lib/python3.8/site-packages/dm/common/client/dmExceptionMapper.py in checkStatus(cls, httpHeaders)
     22             exec('from dm.common.exceptions.%s import %s' % (exModule, exClass))
     23             ex = eval('%s(msg)' % (exClass))
---> 24             raise ex
     25         else:
     26             raise DmException(msg)

DmException: Server raised fault: 'gov.anl.aps.aes.persistence.DAOException: org.hibernate.TransactionException: Transaction not successfully started'

In [19]: 


``

</details>

**Comment by prjemian** [comment]
That's a failure from the data management system when bluesky was asking "What is the name of the current APS run cycle?"  (The correct answer is 2020-2 which goes until the end of FY20.  Only once FY21 starts can the run cycle become 2020-3 for some reason.  Curious that.)

**Comment by prjemian** [comment]
If we have to, we can by pass this for now.  Next change of cycle name is Oct 1.

**Comment by qzhang234** [comment]
Yes, definitely please bypass this. I'm planning on running more tests before 10/01.

Please let me know if this is fixed so that I can close this issue and start other tests next Monday. Thanks!

**Comment by qzhang234** [comment]
Actually @prjemian please let me know if this is fixed as I will be able to test tonight and tomorrow morning too. Thanks a lot!!

**Comment by prjemian** [comment]
Still a problem?

```
(bluesky_2020_9) [quartz ~]$ ipython
Python 3.8.2 (default, Mar 26 2020, 15:53:00) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.16.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import apstools.beamtime.apsbss as bss

In [2]: bss.getCurrentCycle()
Out[2]: '2020-2'

In [3]: 
```

**Comment by prjemian** [comment]
I just fixed this locally (in the local `apstools` package).  There's a PR for this now in `apstools` (https://github.com/BCDA-APS/apstools/pull/418).

Close if you agree this is now resolved.

**Comment by qzhang234** [comment]
Maybe, because my code shows 2020-3

<img width="949" alt="Screen Shot 2020-09-14 at 8 13 28 PM" src="https://user-images.githubusercontent.com/48140482/93153410-ce1a2980-f6c6-11ea-98b0-d8f0a6870cf0.png">

**Comment by prjemian** [comment]
Current cycle is now `2020-2`.  `2020-3` begins Oct 1 at 7 am.  (Weird but true).

**Comment by prjemian** [comment]
Your `2020-3` comes from the directory name.

**Comment by prjemian** [comment]
And that's probably OK.
